app/src/main/java/com/example/myfitnessbuddy/fragments/interior/BuddiesFragment.java

```diff
@@ -342,7 +397,45 @@ public void buddyPictureClicked(int position) {
     public void buddyTrashClicked(int position) {
         buddy = matchedBuddyList.get(position);
         Log.i("Noemi","Remove buddy from list.");
-    }
 
+        databaseReference.addListenerForSingleValueEvent(new ValueEventListener() {
+            @Override
+            public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
+                Log.i("Noemi","Current user: " + currentUserId);
+                if(dataSnapshot.child(Constants.USERS).child(currentUserId).child(Constants.USER_TYPE).getValue().toString().equals(Constants.TRAINEE)){
+                    for(DataSnapshot snapshot : dataSnapshot.child(Constants.MATCHES).getChildren()){
+                        if(snapshot.child(Constants.TRAINEE_ID).getValue().toString().equals(currentUserId) &&
+                                snapshot.child(Constants.TRAINER_ID).getValue().toString().equals(buddy.getId())){
+                            String relaStatus = snapshot.child(Constants.RELATIONSHIP_STATUS).getValue().toString();
+                            snapshot.child(Constants.RELATIONSHIP_STATUS).getRef().setValue(Constants.REMOVED_BY_TRAINEE_STATUS);
+                            //send notification informing user, that he/she is removed
+                            String trainerToken = dataSnapshot.child(Constants.TOKENS).child(buddy.getId()).child(Constants.TOKEN_NODE).getValue().toString();
+                            sendNotifications(trainerToken,Constants.DECLINE_TITLE,Constants.REMOVED_MESSAGE + currentUserId);
+
+                        }
+                    }
+                }
+                else if(dataSnapshot.child(Constants.USERS).child(currentUserId).child(Constants.USER_TYPE).getValue().toString().equals(Constants.TRAINER)){
+                    Log.i("Noemi" , "Here we go: " + currentUserId);
+                    for(DataSnapshot snapshot : dataSnapshot.child(Constants.MATCHES).getChildren()){
+                        if(snapshot.child(Constants.TRAINER_ID).getValue().toString().equals(currentUserId) &&
+                                snapshot.child(Constants.TRAINEE_ID).getValue().toString().equals(buddy.getId())){
+                            String relaStatus = snapshot.child(Constants.RELATIONSHIP_STATUS).getValue().toString();
+                            snapshot.child(Constants.RELATIONSHIP_STATUS).getRef().setValue(Constants.REMOVED_BY_TRAINER_STATUS);
+                            //send notification informing user, that he/she is removed
+//                            String traineeToken = dataSnapshot.child(Constants.TOKENS).child(buddy.getId()).child(Constants.TOKEN_NODE).getValue().toString();
+//                            sendNotifications(traineeToken,Constants.DECLINE_TITLE,Constants.REMOVED_MESSAGE + currentUserId, buddy.getId());
```

Comment:
aici mi se pare destul de mult cod similar. nu am putea face o metoda separata cu  currentUserId, buddy.getId() si snapshot ca parametri? Si sa chemi aceasta metoda din if cu parametri pe care ii folosesti acum in if-ul interior, si chemata din nou in else? Asa acel for loop ar exista o singura data, si la fel if-ul din for ar primi parametri din metoda.